metadata/metadata.go

```diff
@@ -1331,7 +1421,13 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	if has, err := serv.lookup.Has(id); err != nil {
 		return nil, err
 	} else if has {
-		return nil, &ResourceExists{id}
+		existing, err := serv.lookup.Lookup(id)
```

Comment:
I would switch the serv.lookup.Has above into serv.lookup.Lookup
With how this is coded up now, we end up having to do two lookups in the worst case for no reason.
You can check the error of the lookup to make it look like the lookup.Has